library/std/src/path.rs

```diff
@@ -2386,13 +2386,26 @@ impl Path {
         fs::read_dir(self)
     }
 
-    /// Returns `true` if the path points at an existing entity.
+    /// Returns `true` if the path points at an existing entity and syscall succeeds.
     ///
     /// This function will traverse symbolic links to query information about the
     /// destination file. In case of broken symbolic links this will return `false`.
     ///
-    /// If you cannot access the directory containing the file, e.g., because of a
-    /// permission error, this will return `false`.
+    /// # Caveats
```

Comment:
We don't generally call out the caveats as a separate section elsewhere in the documentation of the standard library, as it is very often implied (and similarly, the paragraph about traversing the symbolic links could also be considered a caveat, but is outside of this section).

Reply:
So just simple bold?

> the paragraph about traversing the symbolic links could also be considered a caveat, but is outside of this section

I actually don't find this behavior surprising at all as all other applications behave exactly same. Willing to reconsider if there's a bunch of people who find it surprising.

Reply:
I am on board with a Caveats heading here, it seems appropriate.

Reply:
I am on board with a Caveats heading here, it seems appropriate.